### ISSUE TITLE
Don't require parens on patterns for 0-argument constructors

### ISSUE THREAD

**Issue by Zoetermeer** [Zoetermeer/latro#55]
TLDR, the alpha-converter needs to be augmented such that it can distinguish an id pattern from an ADT pattern (with no parens) on an unqualified 0-argument constructor.  It currently only has two environments, type and value, so it needs some sort of predicate on values that indicates whether said value is a constructor.

Applications involving said constructors will still require parens, but it's annoying and ugly to need them in patterns.  A pattern is not an application at all, anyway, but rather just a pattern that should just look like the alternative declaration (which does not require parens either).

After a first pass at this, it is not as simple as it seems.  An unqualified constructor pattern is indistinguishable from a regular binding-id pattern, but the alpha-converter is unconditionally creating a _new_ binding occurrence for any id pattern of this form.  The typechecker cannot unify because trying to do a lookup on the id in the pattern will never yield the constructor pattern from the environment -- the id in the pattern and the constructor pattern id have been given different unique id's.

Test case:

```
isB : Foo -> Bool                 
isB(A) = False                    
                                  
type Foo = | A | B                
                                  
main(_) = prim(println)(isB(A())) 
```

**Comment by Zoetermeer** [comment]
Resolved in https://github.com/Zoetermeer/latro/commit/8b7780c9dd673a44f505f98904fbeef8e2e31922
